astropy/cosmology/tests/test_cosmology.py

```diff
@@ -1475,9 +1544,14 @@ def test_z_at_value_roundtrip():
         assert allclose(z, funcs.z_at_value(func, fval, zmax=1.5),
                         rtol=2e-8)
 
-    # Test angular_diameter_distance_z1z2
+    # Test distance functions between two redshifts
     z2 = 2.0
-    func = lambda z1: core.Planck13.angular_diameter_distance_z1z2(z1, z2)
-    fval = func(z)
-    assert allclose(z, funcs.z_at_value(func, fval, zmax=1.5),
-                    rtol=2e-8)
+    func_z1z2 = [lambda z1: core.Planck13._comoving_distance_z1z2(z1, z2),
+                 lambda z1: \
```

Comment:
That is an endless debate in software development circles -- should you test non-public interface (white box testing) or not (black box testing)?  Not worth getting worked up over here, however.